tests/AppSwitcher.test.js

```diff
@@ -0,0 +1,234 @@
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+    const neetoUIClassPrefix = "neeto-ui-app-switcher-link--";
+    const activeLink = document.querySelector(`.${neetoUIClassPrefix}active`);
+    expect(activeLink).toBeInTheDocument();
+    expect(getByText(activeLink, "Planner")).toBeInTheDocument();
+  });
+
+  it("should render different sizes correctly", () => {
+    const size = "lg";
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          size={size}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const neetoUIClassPrefix = "neeto-ui-app-switcher__wrapper-size--";
+    expect(
+      document.querySelector(`.${neetoUIClassPrefix}${size}`)
+    ).toBeInTheDocument();
+  });
+
+  it("should render className correctly", () => {
+    const size = "lg";
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          className="test-class"
+          size={size}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    expect(document.querySelector(".test-class")).toBeInTheDocument();
+  });
+
+  it("should close when the back button is pressed", () => {
+    const onClose = jest.fn();
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          onClose={onClose}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const backButton = document.querySelector(
+      "button[data-cy='app-switcher-back-button']"
+    );
+    userEvent.click(backButton);
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  it("should close on pressing escape key", () => {
+    const onClose = jest.fn();
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          closeOnEsc
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          onClose={onClose}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    userEvent.keyboard("{Escape}");
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  it("should close on clicking outside", () => {
+    const onClose = jest.fn();
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          closeOnOutsideClick
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          onClose={onClose}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const appSwitcherBackdrop = document.querySelector(
+      ".neeto-ui-app-switcher__backdrop"
+    );
```

Comment:
Add a `data-testid` to backdrop component and use `screen.getByTestId` to query for backdrop instead.